wagon-providers/wagon-ssh/src/main/java/org/apache/maven/wagon/providers/ssh/jsch/ScpWagon.java

```diff
@@ -297,7 +297,7 @@ public void fillInputData( InputData inputData )
                 throw new IOException( "Invalid transfer header: " + line );
             }
 
-            int filesize = Integer.valueOf( line.substring( 5, index ) ).intValue();
+            long filesize = Long.valueOf( line.substring( 5, index ) );
```

Comment:
Good point. Waiting for this to be integrated and I will ran all tests against the patch.